### PR TITLE
chore: add unregister method

### PR DESCRIPTION
**Pull request by marandaneto**
## :bulb: Motivation and Context
Add `unregister` method.

Follow up https://posthog.slack.com/archives/C04MZFDA5KK/p1709027632605249
the user didn't need to `reset`, but could `register` and `unregister` something.

## :pencil: Checklist

- [X] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] I updated the docs if needed.
- [X] No breaking change or entry added to the changelog.

### CODE REVIEW

File: lib/src/posthog_flutter_web_handler.dart
Comment (marandaneto) [reply lib/src/posthog_flutter_web_handler.dart:91]: Ok looks like there's a bug here.
Calling `register(foo: bar)` and looking at `window.posthog.persistence.props`, I can see `foo: bar`.
Calling `unregister(foo)` removes the `foo` from window.posthog.persistence.props
Calling `capture` again includes the `foo: bar` so something is being cached wrongly.

Comment (daibhin) [reply lib/src/posthog_flutter_web_handler.dart:91]: Strange... sounds like something isn't being called alright

**Review by daibhin — APPROVED** [review]
Your changes look good, so happy if you want to merge this or hold off until the strange behaviour is understood better

**Comment by marandaneto** [comment]
> Your changes look good, so happy if you want to merge this or hold off until the strange behaviour is understood better

will merge it, if there's something off is on the JS SDK, and it works well for Android/iOS (main targets anyway).

**Pull request merged**
